domain.go

```diff
@@ -62,13 +62,55 @@ type DomainInterfaceMAC struct {
 }
 
 type DomainInterfaceModel struct {
-	Type string `xml:type,attr"`
+	Type string `xml:"type,attr"`
+}
+
+type DomainInterfaceSource struct {
+	Bridge  *string `xml:"bridge,attr"`
+	Network *string `xml:"network,attr"`
+	Address *string `xml:"address,attr"`
+	Type    *string `xml:"type,attr"`
+	Path    *string `xml:"path,attr"`
+	Mode    *string `xml:"mode,attr"`
+	Port    *uint   `xml:"port,attr"`
```

Comment:
These shouldn't be using pointers as it makes it tedious to initialize. The style used elswhere is to have omitempty annotations.